### DISCUSSION TITLE
Allow defining and selecting from set-returning functions

### DISCUSSION THREAD

**Discussion by dullbananas** [diesel-rs/diesel#3885] [Ideas]
https://www.postgresql.org/docs/16/functions-srf.html

For example, this could return [1, 4, 9, 16]

```
generate_series(1, 4, 1).select(i * i)
```

**Comment by weiznich** [comment]
For Feature Request proposals we expect quite a lot more details than: "I want to have this or that". 

I would expect at least the following information:

* How does the exact proposed API look like
* What invariants need to be uphold
* What database systems support this syntax
* How would this be implemented (You don't need to provide a complete implementation, but you should at least cover some sketch how something can be represented in the type system)
